src/functions/postResult/framework/handler.ts

```diff
@@ -7,18 +7,29 @@ import { StandardCarTestCATBSchema } from '@dvsa/mes-test-schema/categories/B';
 import { saveTestResult } from '../application/save-result-service';
 import { TestResultDecompressionError } from '../domain/errors/test-result-decompression-error';
 import { bootstrapConfig } from '../../../common/framework/config/config';
+import { validateMESJoiSchema } from '../domain/mes-joi-schema-service';
 
 export async function handler(event: APIGatewayProxyEvent, fnCtx: Context): Promise<Response> {
   await bootstrapConfig();
   console.log(`Invoked with body ${event.body}`);
   const { body } = event;
+  const hasValidationError: boolean = true;
 
   if (isNullOrBlank(body)) {
     return createResponse({}, HttpStatus.BAD_REQUEST);
   }
 
   try {
     const testResult: StandardCarTestCATBSchema = decompressTestResult(body as string);
+    const validationResult = validateMESJoiSchema(testResult);
+
+    if (validationResult.error) {
+      // Validation error thrown with no action possible by examiner - save results in error state - return HTTP 500
+      console.error(`Could not validate the test result body ${validationResult.error}`);
+      await saveTestResult(testResult, hasValidationError);
+      return createResponse({ message: 'Test result body could not be validated' }, HttpStatus.INTERNAL_SERVER_ERROR);
```

Comment:
Need some 4xx series HTTP error code here.

Reply:
I see that this is in-line with the recommendation on the wiki (4xx is only for erroneous results that can be corrected by the examiner - don't know what case that is at this point. Whereas 5xx is cases that can't be resolved by the examiner). Happy to approve on that basis while we figure out the difference.